workflow/executor/os-specific/command_unix.go

```diff
@@ -0,0 +1,75 @@
+//go:build linux || darwin
+
+package os_specific
+
+import (
+	"io"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+
+	"github.com/creack/pty"
+	"golang.org/x/term"
+)
+
+func StartCommand(cmd *exec.Cmd, stdin *os.File, stdout io.Writer, stderr io.Writer) (func(), error) {
```

Comment:
It seems weird to me that we pass stdin/stdout/stderr twice. Once as an explicit parameter, once implicitly as cmd.Std*. Why not just the latter?

Reply:
Could you nil them in the function? Otherwise you’re leaking pty implementation detail up the stack.